templates/common/__test__/get.location.test.ts

```diff
@@ -0,0 +1,50 @@
+import assert from 'node:assert';
+import fs from 'node:fs';
+import { describe, it } from 'node:test';
+
+import { runTestFunction } from './function.helper.js';
+
+describe('get-location script template', () => {
+  it('should output workflow artifact location', (t) => {
+    const spy = t.mock.method(console, 'log');
+    const shimRequired: string[] = [];
+    runTestFunction(
+      './templates/common/get.location.yml',
+      [
+        {
+          toReplace: `JSON.parse(process.env['ARGO_TEMPLATE'])`,
+          replaceWith: JSON.stringify({
+            archiveLocation: {
+              s3: {
+                key: '2024-10/02-test-get-location-29l4x/test-get-location-29l4x-get-location-3125883809',
+                bucket: 'linz-workflows-scratch',
+              },
+            },
+          }),
+        },
+      ],
+      (req) => {
+        shimRequired.push(req);
+        if (req === 'node:fs') return fs;
+        throw new Error('Failed');
+      },
+    );
+    assert.equal(spy.mock.callCount(), 1);
+
+    const logOutputDict = JSON.parse(String(spy.mock.calls[0]?.arguments[0])) as { time: number };
+
+    // override time
+    logOutputDict.time = 1724037007216;
+
+    assert.deepEqual(logOutputDict, {
+      time: 1724037007216,
```

Comment:
It might be easier (avoiding a magic number) to assert that it's equal to itself rather than overwriting it:

```suggestion
    const logOutputDict = JSON.parse(String(spy.mock.calls[0]?.arguments[0]));

    assert.deepEqual(logOutputDict, {
      time: logOutputDict.time,
```